thorlcr/activities/thdiskbase.cpp

```diff
@@ -121,9 +121,17 @@ void CDiskReadMasterBase::serializeSlaveData(MemoryBuffer &dst, unsigned slave)
         CSlavePartMapping::serializeNullMap(dst);
 }
 
+void CDiskReadMasterBase::getActivityStats(IStatisticGatherer & stats)
+{
+    CMasterActivity::getActivityStats(stats);
+    diskAccessCost = calcFileReadCostStats(false);
+    if (diskAccessCost)
+        stats.addStatistic(StCostFileAccess, diskAccessCost);
+}
+
 void CDiskReadMasterBase::done()
 {
-    updateFileReadCostStats();
+    diskAccessCost = calcFileReadCostStats(true);
```

Comment:
In the long run, we should be publishing all stats. periodically, i.e. also publishing the file access stats as a Thor subgraph is running, not just at the end.

And it would be okay to do so now for longer running subgraphs, but it could be painful for frequent short running subgraphs.

This is related to https://track.hpccsystems.com/browse/HPCC-29207.
We will need to revisit this/other areas which delay updates like this when HPCC-29207 is worked on.